### ISSUE TITLE
NSGA-Net: Neural Architecture Search using Multi-Objective Genetic Algorithm

### ISSUE THREAD

**Issue by sile** [sile/readings#124]
URL
-----

https://arxiv.org/abs/1810.03522

Abstract
----------

> This paper introduces NSGA-Net -- an evolutionary approach for neural architecture search (NAS). NSGA-Net is designed with three goals in mind: (1) a procedure considering multiple and conflicting objectives, (2) an efficient procedure balancing exploration and exploitation of the space of potential neural network architectures, and (3) a procedure finding a diverse set of trade-off network architectures achieved in a single run. NSGA-Net is a population-based search algorithm that explores a space of potential neural network architectures in three steps, namely, a population initialization step that is based on prior-knowledge from hand-crafted architectures, an exploration step comprising crossover and mutation of architectures, and finally an exploitation step that utilizes the hidden useful knowledge stored in the entire history of evaluated neural architectures in the form of a Bayesian Network. Experimental results suggest that combining the dual objectives of minimizing an error metric and computational complexity, as measured by FLOPs, allows NSGA-Net to find competitive neural architectures. Moreover, NSGA-Net achieves error rate on the CIFAR-10 dataset on par with other state-of-the-art NAS methods while using orders of magnitude less computational resources. These results are encouraging and shows the promise to further use of EC methods in various deep-learning paradigms.

Table of Contents
--------------------

- (1) Introduction
  - (Feature 1) Multi-Objective Optimization
  - (Feature 2) Flexible Architecture Search Space
  - (Feature 3) Non-Dominated Sorting
  - (Feature 4) Efficient Recombination
  - (Feature 5) Bayesian Learning
- (2) Related Work
  - Reinforcement Learning
  - Evolutionary Algorithms
  - Other Methods
- (3) Proposed Approach
  - (3-1) Encoding
    - Operation Encoding
    - Search Space
  - (3-2) Search Procedure
    - Exploration
    - Crossover
    - Mutation
    - Exploitation
- (4) Experiments
  - (4-1) Performance Metrics
  - (4-2) Implementation Details
    - Dataset
    - NSGA-Net hyper-parameters
    - Network training during searching
  - (4-3) Architecture Validation
  - (4-4) Results Analysis
  - (4-5) Transferability
  - (4-6) Ablation Studies
    - Crossover Operator
    - Bayesian Network (BN) based Offsprint Creation
  - (4-7) Discussion
- (5) Conclusions
- (6) Appendix
  - (6-1) Duplicate Checking and Removal
  - (6-2) Architecture Complexity Estimation

BibTeX
------

```bibtex
@misc{lu2018nsganet,
    title={NSGA-Net: Neural Architecture Search using Multi-Objective Genetic Algorithm},
    author={Zhichao Lu and Ian Whalen and Vishnu Boddeti and Yashesh Dhebar and Kalyanmoy Deb and Erik Goodman and Wolfgang Banzhaf},
    year={2018},
    eprint={1810.03522},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```

See Also
----------

- GitHub: https://github.com/ianwhale/nsga-net
